pinot-query-runtime/src/test/resources/queries/WithStatements.json

```diff
@@ -0,0 +1,81 @@
+{
+  "with_statement_tests": {
+    "psql": "7.8.1",
+    "tables": {
+      "tbl1": {
+        "schema": [
+          {"name": "strCol", "type": "STRING"},
+          {"name": "intCol", "type": "INT"}
+        ],
+        "inputs": [
+          ["a", 3],
+          ["b", 2],
+          ["c", 5],
+          ["a", 1]
+        ]
+      },
+      "tbl2": {
+        "schema": [
+          {"name": "strCol1", "type": "STRING"},
+          {"name": "strCol2", "type": "STRING"},
+          {"name": "intCol", "type": "INT"}
+        ],
+        "inputs": [
+          ["a", "foo", 1],
+          ["a", "bar", 2],
+          ["b", "alice", 42],
+          ["b", "bob", 196883]
+        ]
+      }
+    },
+    "queries": [
+      {
+        "description": "single with with group by",
+        "sql": "WITH w AS ( SELECT strCol, SUM(intCol) AS sumRes FROM {tbl1} GROUP BY strCol ) SELECT strCol, strCol2, sumRes FROM w JOIN {tbl2} ON w.strCol = {tbl2}.strCol1",
```

Comment:
Can we add an example where the `JOIN` in outer query is done between `WITH` virtual table W and the same table (`tbl1`) used in `WITH` ?

Reply:
Basically WITH can serve for decoration as well (using self-join) where we want to enhance the result of a standard GROUP BY aggregation query on a table with additional columns from the same table.

Something like ....

```
WITH maxResult AS (
  SELECT
    groupCol1,
    groupCol2,
    MAX(aggCol) AS M
  FROM T
  GROUP BY groupCol1, groupCol2
)
SELECT T.*, maxResult.M
 FROM T
 JOIN 
 maxResult
 ON 
 T.groupCol1 = maxResult.groupCol1
 AND 
 T.groupCol2 = maxResult.groupCol2
```

Reply:
Can we also add one example for transform inside agg.  ? 

Here is a useful example in the context of a non JOIN query...

```
WITH metricResult AS ( 
      SELECT 
         groupCol1, 
         groupCol2, 
         SUM(c1 * c2) AS metric 
      FROM T 
      GROUP BY groupCol1, groupCol2
) 
SELECT 
  groupCol1,
  MAX(metricResult.metric)
  FROM metricResult
  GROUP BY groupCol1
  ORDER BY MAX(metricResult.metric) DESC
```

Reply:
this is added